.github/actions/make_init/action.yml

```diff
@@ -0,0 +1,64 @@
+name: "Setup & Initialize Virtual Env"
+runs:
+  using: composite
+  steps:
+    # - name: Restore pre-commit cache
+    #   id: cache-pre-commit
+    #   uses: actions/cache@v3
+    #   with:
+    #     path: ~/.cache
+    #     key: v1-pre-commit-${{ hashFiles('**/.pre-commit-config.yaml') }}
+    # - if: steps.cache-pre-commit.outputs.cache-hit != 'true'
+    #   name: Install pre-commit
+    #   run: |
+    #     sudo pre-commit install-hooks
+    #   shell: bash -ileo pipefail {0}
+    - name: Setup Node
+      uses: actions/setup-node@v3
+      with:
+        node-version-file: '.nvmrc'
+        cache: 'yarn'
+        cache-dependency-path: '**/yarn.lock'
+    - name: Initialize React
+      run: make react-init
+      shell: bash -ileo pipefail {0}
+    - name: Install Dependencies
+      run: |
+        sudo apt update
+        # protobuf, pyodbc, dot, & graphviz dependencies
+        sudo apt install -y gnupg \
+            unixodbc-dev \
+            graphviz \
+            libgvc6 \
+            python3-pydot \
+            python3-pydot-ng \
+
+        echo "deb http://ppa.launchpad.net/maarten-fonville/protobuf/ubuntu trusty main" | sudo tee /etc/apt/sources.list.d/protobuf.list
+        sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 4DEA8909DC6A13A3
+        sudo apt update
+        sudo apt install -y protobuf-compiler
+      shell: bash -ileo pipefail {0}
+    - name: Restore virtualenv from cache
+      id: cache-virtualenv
+      uses: actions/cache@v3
+      with:
+        path: venv
+        key: v1-python-venv-${{ hashFiles('**/python_cache_key.md5') }}
+    - if: steps.cache-virtualenv.outputs.cache-hit != 'true'
+      name: Create Virtual Env
+      run: |
+        python -m venv venv
+        source venv/bin/activate
+        pip install --upgrade pip
+        make setup
+        make pipenv-install
+        deactivate
+
+        # Add 'activate venv' to $HOME/.bash_profile.
+        echo 'source venv/bin/activate' >> $HOME/.bash_profile
```

Comment:
```suggestion
        echo "PATH=${PWD}/venv/bin:${PATH}" >> ${GITHUB_ENV}
        echo "VIRTUAL_ENV=${PWD}/venv" >> ${GITHUB_ENV}
```
Can you see if this will solve your problems with the virtual environment?
Here is a related Dockerfile:
https://github.com/streamlit/streamlit/blob/67cb14159112aaabc1369cc408099868bd3267e2/e2e/Dockerfile#L86-L89
Here is doc about `${GITHUB_ENV}` env variable:
https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-environment-variable

Reply:
I made the suggested change and tested a couple times with the same result 😓

Reply:
It works for me. See:
https://github.com/streamlit/streamlit/compare/git-jstests...sfc-gh-kbregula-git-jstests
and
https://github.com/streamlit/streamlit/runs/7888405351?check_suite_focus=true

Reply:
I admittedly hadn't updated the shell config across files to the `shell: bash -euxo pipefail {0}` in the last set of trials...
It now works for running the JS test (run [here](https://github.com/streamlit/streamlit/actions/runs/2878921933)) but still causes issues with the python test (run [here](https://github.com/streamlit/streamlit/actions/runs/2878921934)) without the sourcing - specifically erroring at the click module.

Reply:
Not sure if this comment slipped by, but I'm fairly certain that the problem is due to the lines setting the env / editing `.bash_profile` only running on a cache miss: https://github.com/streamlit/streamlit/pull/5158#discussion_r948483186

Reply:
Vincent you're correct & my hero, the problem is resolved by adding `echo 'source venv/bin/activate' >> $HOME/.bash_profile` to a separate step.